tfhe/src/high_level_api/integers/tests.rs

```diff
@@ -817,3 +817,71 @@ fn test_if_then_else() {
         if clear_a <= clear_b { clear_b } else { clear_a }
     );
 }
+
+#[test]
+fn test_integer_key_switching() {
+    let config = ConfigBuilder::all_disabled()
+        .enable_default_integers()
+        .build();
+    let (client_key_1, server_key_1) = generate_keys(config.clone());
+    let (client_key_2, server_key_2) = generate_keys(config);
+    let ksk = KeySwitchingKey::for_same_parameters(
+        (&client_key_1, &server_key_1),
+        (&client_key_2, &server_key_2),
+    )
+    .unwrap();
+
+    let clear = 12_837u16;
+    let a = FheUint16::encrypt(clear, &client_key_1);
+
+    let a = a.keyswitch(&ksk);
+    let da: u16 = a.decrypt(&client_key_2);
+    assert_eq!(da, clear);
+}
+
+#[test]
+fn test_integer_key_switching_2() {
+    let mut parameters = crate::shortint::parameters::PARAM_MESSAGE_1_CARRY_1_KS_PBS;
+    parameters.message_modulus.0 = 4;
+    parameters.carry_modulus.0 = 4;
```

Comment:
this is beyond weird and likely incorrect ? or are we getting away with "murder" by just encrypting ? plus integer keyswitching between different params would require resizing blocks and shuffling stuff around right in the general case ?

Reply:
This is what JB told me to do (or at least what I understand I should do for the test)

Reply:
yes ok, I'm just not clear on what the consequence is for correctness of the encryption KS (PBS ?) decryption we are testing here

does this tests pass let's say 1000 iterations with different inputs/keys ?